sale_channel_hook/models/sale_channel.py

```diff
@@ -0,0 +1,47 @@
+#  Copyright (c) Akretion 2020
+#  License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl.html)
+
+import hashlib
+import hmac
+
+import requests
+
+from odoo import _, fields, models
+from odoo.exceptions import ValidationError
+
+from odoo.addons.queue_job.job import job
+
+
+class SaleChannel(models.Model):
+    _inherit = "sale.channel"
+
+    auth_token = fields.Char("Secret authentication token")  # DISCUSSION: sécurité
+    api_endpoint = fields.Char("Hooks API endpoint")
+
+    @job
+    def send_hook_api_request(self, hook_name, content):
+        # check necessary info is filled
+        if not self.api_endpoint or not self.auth_token:
+            raise ValidationError(
+                _(
+                    "Set a secret token and define an API "
+                    "endpoint to use this channel's hook"
+                )
+            )
+        # check hook is activated
```

Comment:
done. 

Check that the hook is active will be made on the object using the hook (delivery, sale, invoice) instead of the sale channel's job to avoid creating useless jobs